### DISCUSSION TITLE
Custom UrlDecoder

### DISCUSSION THREAD

**Discussion by ItSkary** [ghorsey/OpenGraph-Net#34] [Q&A]
I was reading the project source before incorporating (through nuget) into the product i'm working on.
I have to admint that at this moment i have just a grasp about how this project work, but lot of thing still missing in the big picture in my mind.

A thing that certainly surprise me, is that i read this method that seems to implement a custom url decoder : 

In OpenGraph.cs
```
    /// <summary>
    /// Safes the HTML decode URL.
    /// </summary>
    /// <param name="value">The value.</param>
    /// <returns>The string.</returns>
    private static string HtmlDecodeUrl(string value)
    {
        if (value == null)
        {
            return string.Empty;
        }

        // naive attempt
        var patterns = new Dictionary<string, string>
        {
            ["&amp;"] = "&",
        };

        foreach (var key in patterns)
        {
            value = value.Replace(key.Key, key.Value);
        }

        return value;
    }
```

I am really surprised about this code, why doesn't use the framework objects to do so? And why writing a decoding method that only handle '&amp;'. I am really puzzled about that.

**Comment by ghorsey** [comment]
Good question. It's been a long time since I looked through all this code! 

IIRC, this is carried over from the original .Net Framework version of this library.  Back in the day to access html en/decode you had to go through `HttpContext` like `HttpContext.Current.Server.HtmlDecode()`. I believe this was before `HttpUtility` was a thing.

Since this was a class library, I couldn't guarantee that `HttpContext` was present and this was the simplest solution for what was needed at the time.

So today, this is likely a bit of technical debt.

**Reply by ghorsey** [reply]
Also, if you want, fix this bit of code to use `HttpUtility` and raise a PR if all the tests continue to pass 😃
